@xen-orchestra/lite/src/components/host/network/HostPifStatus.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <UiInfo class="pif-status text-ellipsis" :accent="getStatusProps(status).accent">
+    <p class="text-ellipsis" :class="{ card }">{{ getStatusProps(status).text }}</p>
+  </UiInfo>
+</template>
+
+<script setup lang="ts">
+import UiInfo from '@core/components/ui/info/UiInfo.vue'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { faCheck, faExclamation } from '@fortawesome/free-solid-svg-icons'
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { pifStatus, card } = defineProps<{
+  pifStatus: any
+  card?: boolean
+}>()
+
+const { t } = useI18n()
+
+type Status = 'connected' | 'disconnected' | 'partial'
+type Accent = 'success' | 'warning' | 'danger'
+
+const states = computed<Record<Status, { text: string; icon: IconDefinition; accent: Accent }>>(() => ({
+  connected: { text: t('connected'), icon: faCheck, accent: 'success' },
+  disconnected: { text: t('disconnected'), icon: faCheck, accent: 'danger' },
+  partial: { text: t('disconnected-from-physical-device'), icon: faExclamation, accent: 'warning' },
+}))
+
+const status = computed(() => {
+  if (pifStatus) {
+    return 'connected'
+  }
+  return 'disconnected'
+})
+
+const getStatusProps = (status: Status) => states.value[status]
```

Comment:
I don't understand this function.

Looking at the template, we can see it only receives the `status` computed property, so it could be a computed property itself:

```suggestion
const statusProps = computed(() => states.value[status.value])
```

However, the `status` computed property only returns either 'connected' or 'disconnected', while the `states` object contains three properties: 'connected', 'disconnected', and 'partial'.

This means the 'partial' property appears to never be used.

Furthermore, we can see that the `status` computed property is only used within the `statusProps` computed property. This creates redundancy, suggesting that all this logic could likely be simplified by combining it into a single computed property.

Reply:
Yes, sure